backend/src/utils/resourceUtils.ts

```diff
@@ -615,31 +624,182 @@ export const getConsoleLinks = (): ConsoleLinkKind[] => {
+        e.response?.body?.message || e?.response?.statusMessage
+      }`;
+    });
+
+const createSuccessfulMigrationConfigMap = async (
+  fastify: KubeFastifyInstance,
+  configMapName: string,
+  description: string,
+): Promise<void> => {
+  // Create configmap to flag operation as successful
+  const configMap: V1ConfigMap = {
+    metadata: {
+      name: configMapName,
+      namespace: fastify.kube.namespace,
+    },
+    data: {
+      migratedCompleted: 'true',
+    },
+  };
+  return await fastify.kube.coreV1Api
+    .createNamespacedConfigMap(fastify.kube.namespace, configMap)
+    .then(() => fastify.log.info(`Successfully migrated ${description}`))
+    .catch((e) => {
+      throw `A ${
+        e.statusCode
+      } error occurred when trying to create configmap for ${description} migration: ${
+        e.response?.body?.message || e?.response?.statusMessage
+      }`;
     });
+};
+
+export const cleanupKserveRoleBindings = async (fastify: KubeFastifyInstance): Promise<void> => {
+  // When we startup — in kube.ts we can handle a migration (catch ALL promise errors — exit gracefully and use fastify logging)
+  // Check for migration-kserve-inferenceservices-role configmap in dashboard namespace — if found, exit early
+  const CONFIG_MAP_NAME = 'migration-kserve-inferenceservices-role';
+  const DESCRIPTION = 'KServe secure rolebindings';
+
+  const continueProcessing = await shouldMigrationContinue(fastify, CONFIG_MAP_NAME, DESCRIPTION);
+
+  if (continueProcessing) {
+    const roleBindings = await fastify.kube.customObjectsApi.listClusterCustomObject(
+      'rbac.authorization.k8s.io',
+      'v1',
+      'rolebindings',
+      undefined,
+      undefined,
+      undefined,
+      `${KnownLabels.DASHBOARD_RESOURCE} = true`,
+    );
+    const kserveSARoleBindings =
+      (roleBindings?.body as V1RoleBindingList).items?.filter(
+        ({ roleRef, subjects, metadata }) =>
+          roleRef.kind === 'ClusterRole' &&
+          roleRef.name === 'view' &&
+          subjects.length === 1 &&
+          subjects[0].kind === 'ServiceAccount' &&
+          metadata?.ownerReferences.length === 1 &&
+          metadata?.ownerReferences[0].kind === 'InferenceService',
+      ) || [];
+
+    const replaceRoleBinding = async (existingRoleBinding: V1RoleBinding) => {
+      const inferenceServiceName = existingRoleBinding.metadata?.ownerReferences[0].name;
+      const namespace = existingRoleBinding.metadata?.namespace;
+      const newRoleBindingName = `${inferenceServiceName}-view`;
+      const newRoleName = `${inferenceServiceName}-view-role`;
+
+      const newRole: V1Role = {
+        apiVersion: 'rbac.authorization.k8s.io/v1',
+        kind: 'Role',
+        metadata: {
+          name: newRoleName,
+          namespace,
+          labels: {
+            [KnownLabels.DASHBOARD_RESOURCE]: 'true',
+          },
+        },
+        rules: [
+          {
+            verbs: ['get'],
```

Comment:
To add this role we need to add to the serviceaccount cluster role the following capability:

```
  - apiGroups:
      - serving.kserve.io
    resources:
      - inferenceservices
    verbs:
      - get
      - list
      - watch
```

Reply:
@lucferbux sorry for the delay here - just made this change